src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SKIPPED);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsDetailsToReportByResultCode(List<RassBusinessObjectUpdateResult<R>> results,
+            Class businessObjectClass, RassObjectUpdateResultCode resultCode) {
+        List<RassBusinessObjectUpdateResult<R>> filteresdResults = results.stream().filter(result -> resultCode.equals(result.getResultCode()))
+                .collect(Collectors.toList());
+
+        writeHeaderForResultsDetailsByResultCode(businessObjectClass, resultCode);
+
+        for (RassBusinessObjectUpdateResult<R> result : filteresdResults) {
+            writeResultDetailToReport(result);
+        }
+    }
+
+    private void writeHeaderForResultsDetailsByResultCode(Class businessObjectClass, RassObjectUpdateResultCode resultCode) {
+        String header = StringUtils.EMPTY;
+        String objectName = businessObjectClass.getSimpleName();
+        switch (resultCode) {
+            case SUCCESS_NEW:
+                header = "** " + objectName + " objects successfully created:";
+                break;
+            case SUCCESS_EDIT:
+                header = "** " + objectName + " objects successfully updated:";
+                break;
+            case ERROR:
+                header = "** " + objectName + " objects that had errors:";
+                break;
+            case SKIPPED:
+                header = "** " + objectName + " objects that were skipped:";
+                break;
+        }
+        reportWriterService.writeNewLines(1);
+        reportWriterService.writeFormattedMessageLine(header);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultDetailToReport(RassBusinessObjectUpdateResult<R> result) {
+        String reportMessage = StringUtils.EMPTY;
+        switch (result.getResultCode()) {
+            case SUCCESS_NEW:
+                reportMessage = result.getBusinessObjectClass().getSimpleName() + " #: " + result.getPrimaryKey() + " created by document #: "
+                        + result.getDocumentId();
+                break;
+            case SUCCESS_EDIT:
+                reportMessage = result.getBusinessObjectClass().getSimpleName() + " #: " + result.getPrimaryKey() + " updated by document #: "
+                        + result.getDocumentId();
+                break;
+            case ERROR:
+                reportMessage = result.getBusinessObjectClass().getSimpleName() + " #: " + result.getPrimaryKey() + " resulted in error. \n Error(s): "
+                        + result.getErrorMessage();
+                break;
+            case SKIPPED:
+                reportMessage = result.getBusinessObjectClass().getSimpleName() + " #: " + result.getPrimaryKey() + " skipped.";
+                break;
+        }
```

Comment:
Another switch block where it would be helpful to have a "default" case.

Also, all of these particular cases are duplicating the same prefix. You may want to generate the common part of the message outside of the switch block and then concatenate things accordingly.

Reply:
updated